scripts/common.sh

```diff
@@ -61,3 +61,72 @@ restart_agents() {
         exec_pod $POD "elastic-agent restart"
     done
 }
+
+bump_preview_version() {
+    local current_version="${1:?Missing current version}"
+    preview_number="${current_version##*-preview}"
+    ((next_preview_number = 10#${preview_number} + 1))
+    echo "${current_version%-*}-preview$(printf "%02d" $next_preview_number)"
+}
+
+bump_minor_version() {
+    local version="${1:?Missing version}"
+    IFS='.' read -r major minor _ <<<"$1"
+    ((minor++))
+    echo "$major.$minor.0"
+}
+
+get_integration_version() {
+    local changelog_path="${1:?Missing changelog.yml path}"
+    current_version=$(yq '.[0].version' "$changelog_path" | tr -d '"')
+    echo "$current_version"
+}
+
+get_new_integration_version_map_entry() {
+    local latest_version="${1:?Missing latest versions entry}"
+    IFS='-' read -r group1 group2 <<<"$latest_version"
+    IFS='.' read -r major1 minor1 _ <<<"$group1"
+    IFS='.' read -r major2 minor2 _ <<<"$group2"
+    ((minor1++))
+    ((minor2++))
+    local first_version="${major1}.${minor1}.x"
+    local second_version="$(echo "${major2}.${minor2}.x" | xargs)"
+    echo "$first_version - $second_version"
+}
+
+# bumps existing preview version: 1.0.0-preview01 -> 1.0.0-preview02, or
+# creates a new preview version: 1.0.0 -> 1.1.0-preview01, and
+# updates the manifest and changelog files
+bump_integration_version() {
```

Comment:
a PR to an integration must include a changelog and manifest update. depending on when the PR is made, the current integration version may be one of the following:
1. some preview version, like `1.0.0-preview01`, in which case the preview suffix will be incremented. see [example PR](https://github.com/elastic/integrations/pull/9358/files) for actual changes made
2. some version, like `1.0.0`, in which case the changes will result in this example diff:
```diff
+++ b/packages/cloud_security_posture/changelog.yml
@@ -1,5 +1,6 @@
 # newer versions go on top
 # version map:
+# 1.10.x - 8.15.x
 # 1.9.x - 8.14.x
 # 1.8.x - 8.13.x
 # 1.7.x - 8.12.x
@@ -8,6 +9,11 @@
 # 1.4.x - 8.9.x
 # 1.3.x - 8.8.x
 # 1.2.x - 8.7.x
+- version: 1.10.0-preview01
+  changes:
+    - description: Bump version
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/9328
 - version: "1.9.0"
   changes:
     - description: Convert fields to secrets
diff --git a/packages/cloud_security_posture/manifest.yml b/packages/cloud_security_posture/manifest.yml
index bda29ce3e..aaa4fcca8 100644
--- a/packages/cloud_security_posture/manifest.yml
+++ b/packages/cloud_security_posture/manifest.yml
@@ -1,7 +1,7 @@
 format_version: 3.0.0
 name: cloud_security_posture
 title: "Security Posture Management"
-version: "1.9.0"
+version: "1.10.0-preview01"
 source:
   license: "Elastic-2.0"
 description: "Identify & remediate configuration risks in your Cloud infrastructure"
@@ -11,7 +11,7 @@ categories:
   - cloudsecurity_cdr
 conditions:
   kibana:
-    version: "^8.14.0"
+    version: "^8.15.0"
   elastic:
     subscription: basic
     capabilities:
```